src/librustc_lint/types.rs

```diff
@@ -680,26 +692,38 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
 
                         // Check for a repr() attribute to specify the size of the
                         // discriminant.
-                        if !def.repr.c() && def.repr.int.is_none() {
+                        if !def.repr.c() && !def.repr.transparent() && def.repr.int.is_none() {
                             // Special-case types like `Option<extern fn()>`.
                             if !is_repr_nullable_ptr(cx, def, substs) {
                                 return FfiUnsafe {
                                     ty: ty,
                                     reason: "enum has no representation hint",
-                                    help: Some("consider adding a #[repr(...)] attribute \
-                                                to this enum"),
+                                    help: Some("consider adding a `#[repr(C)]`, \
+                                                `#[repr(transparent)]`, or integer `#[repr(...)]` \
+                                                attribute to this enum"),
```

Comment:
This is likely an improvement over the previous suggestion, but given how niche transparent enums (& unions, above) are it exacerbates #60757. Not a blocker, just cross-referencing.

Reply:
Glad you pointed it out. I'm going to mark it as resolved since there's not much I plan on doing about this, but I totally agree.